source/extensions/filters/network/thrift_proxy/auto_protocol_impl.cc

```diff
@@ -41,7 +45,19 @@ bool AutoProtocolImpl::readMessageBegin(Buffer::Instance& buffer, MessageMetadat
 
     uint16_t version = buffer.peekBEInt<uint16_t>();
     if (BinaryProtocolImpl::isMagic(version)) {
-      setType(ProtocolType::Binary);
+      // 12 bytes is the minimum length for message-begin in the binary protocol.
+      if (buffer.length() < 12) {
```

Comment:
Nit: raw 12 here feels icky, unclear how it relates to following code and its expectations. It might not be much clearer to add a constant at top of file, but for your consideration.